sonar-plugin-api/src/main/java/org/sonar/api/server/rule/internal/DefaultNewRule.java

```diff
@@ -388,11 +392,10 @@ public DefaultNewRule addDeprecatedRuleKey(String repository, String key) {
   }
 
   @Override
-  public DefaultNewRule addEducationPrincipleKeys(String... list) {
-    for (String educationPrincipleKey : list) {
-      EducationPrincipleKeyFormat.validate(educationPrincipleKey);
-      this.educationPrincipleKeys.add(educationPrincipleKey);
-    }
+  public DefaultNewRule addEducationPrincipleKeys(String... keys) {
+    Set<String> candidateEducationPrincipleKeys = Arrays.stream(keys).collect(Collectors.toSet());
```

Comment:
Maybe a bit shorter with `Set.of(keys)` ?

EDIT: after suggesting this I tried to search if we use Java 8 or 11 to build the API but I was not able to find the info. We better specify a Gradle toolchain to make it explicit and consistent for devs and CI

Reply:
fixed :)